### PR TITLE
workaround: notifications not displaying for WhatsApp

### PR DESCRIPTION
**Pull request by agreenbhm**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change
Added try/catch for displaying notifications, with notification options object setting onClick to null in the event of error.  Fixes lack of notifications for WhatsApp (and possibly others) due to failure to clone the options object while displaying notifications (related to onClick function).  This is a workaround until the full problem is resolved, as the change allows notifications to be displayed but when clicking them you are not taken to the proper screen (due to null'ing the onClick function).  This is much better than not receiving any notifications while waiting for a full fix, IMO.

#### Motivation and Context
Workaround/Fix for https://github.com/ferdium/ferdium-app/issues/1105 - WhatsApp notifications are not being displayed.

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes

Workaround for notifications not displaying for some apps (WhatsApp + more).

### CODE REVIEW

File: src/webview/notifications.ts
Comment (mcmxcdev) [inline src/webview/notifications.ts:55]: I think you got confused a bit with the try catch and then/catch blocks, With your code, the same method would be called in both try and catch. 

What do you think about this:

```suggestion
try {
  window.ferdium.displayNotification(title, options);

  if (typeof this.onClick === 'function') {
    this.onClick();
  }
} catch {
  this.options.onClick = null;
}
```

**Review by mcmxcdev — COMMENTED** [review]
Not sure if we want to merge this including the "null'ing the onClick function" workaround since it doesn't redirect to the correct screen as you mentioned. 

What do you think @vraravam ?

File: src/webview/notifications.ts
Comment (agreenbhm) [reply src/webview/notifications.ts:55]: I understand why it is seen as a bit repetitive, however I think it may be necessary.  IIRC, 'this.onClick' will still return as being typeof function even in the failure scenario, but when the displayNotifications function is called the "options" object seems to fail to be copied for some reason (something seemingly to do with the onClick function).  My point is, the error occurs further down the line so we can't just assume that it showing as typeof function is enough for it to succeed.

Comment (agreenbhm) [reply src/webview/notifications.ts:55]: Just checked, and as I said, it seems that typeof returns the correct value even if it will then go on to fail.  See screenshot below.  First debug console command was run at the breakpoint, and then when execution is resumed the error is thrown.

image

Comment (mcmxcdev) [reply src/webview/notifications.ts:55]: Thanks for the thorough investigation! It seems it has to be written that way, so fine with continuing. 

Still would like to defer to @vraravam

Comment (vraravam) [reply src/webview/notifications.ts:55]: I am not enough of a js developer to comment on this nuance. I will defer to you and other core committers for the decision.

Comment (agreenbhm) [reply src/webview/notifications.ts:55]: I'm no JS expert either. That said, this seems to mostly fix a significant issue even if it's a bit of a dirty hack, so I hope you will approve it.

**Review by mcmxcdev — APPROVED** [review]

File: src/webview/notifications.ts
Comment (SpecialAro) [inline src/webview/notifications.ts:55]: Any reason we are having a try catch block instead of (async function).then().catch() ?

I agree everything is a bit of an hack and we should probably make some sense of it in the future.

EDIT: also, just noticed we are repeating code in the catch section. Maybe we should move that logic to its own function, no?

**Pull request merged**
